shell/src-app/Linter.hs

```diff
@@ -0,0 +1,83 @@
+-- Allow ourselves to write Text literals.
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+    import Prologue hiding (Symbol)
+    import Luna.Syntax.Text.Lexer.Runner
+    import Luna.Syntax.Text.Lexer.Token as T
+    import Luna.Syntax.Text.Lexer.Symbol as S
+    import Data.Text.Position            (Delta)
+    import Data.Text32 hiding (convert)
+    import Data.Aeson
+    import qualified Data.ByteString.Lazy as B
+
+    instance ToJSON Delta where
+        toJSON delta = toJSON ((convert delta) :: Int)
+
+    jsonStr :: String -> Value
+    jsonStr = toJSON
+
+    jsonT32 :: Text32 -> Value
+    jsonT32 txt = toJSON ((convert txt) ::String)
+
+    symbolWithValue :: String -> Text32 -> Value
+    symbolWithValue name value = object 
+        [ "symbol" .= toJSON name
+        , "value" .= jsonT32 value ]
+
+    symbolKw :: String -> Value
+    symbolKw name = object 
+        [ "symbol" .= jsonStr "Keyword"
+        , "value" .= toJSON name ]
+
+    instance ToJSON Symbol where
```

Comment:
Why don't you derive this instance? Making instances by hand is rather a bad idea in almost every case.

Reply:
Can you explain this further? I don't exactly understand what you mean (my experience with Haskell's typeclasses is quite slim)

Reply:
Sure. Haskell can write instances for you. See the example code in the docs: http://hackage.haskell.org/package/aeson-1.4.2.0/docs/Data-Aeson.html 

Especially:
```haskell
instance ToJSON Person where
    -- No need to provide a toJSON implementation.
```

Reply:
Read more about it here: https://wiki.haskell.org/GHC/Stand-alone_deriving_declarations
and here: https://wiki.haskell.org/Generics

Reply:
This is the beauty of Algebraic Datatypes in languages like Haskell or Luna ;)

Reply:
Ah now I more or less understood what you meant.
It's because I wanted custom behaviour here - to get tags etc. It will be visible in the new commit.
```haskell
    instance ToJSON Symbol where
        toJSON :: Symbol -> Value
        toJSON sym = case valueForSymbol sym of
            Nothing -> object
                [ "symbol" .= text32ToJSON (showCons sym)
                , "tags" .= map text32ToJSON (getTags sym) ]
            Just v -> object
                [ "symbol" .= text32ToJSON (showCons sym)
                , "tags" .= map text32ToJSON (getTags sym)
                , "value" .= v 
                ]
```

Reply:
Then create another datatype which wraps over tags and Symbol and derive instance for it as well. Manual way is error prone and you should not use it in such case.

Reply:
Unfortunately when using the deriving mechanism, all keys get underscores (as that's how they are defined there (because of Lenses?)).